RecentActivity/src/org/sleuthkit/autopsy/recentactivity/Extract.java

```diff
@@ -216,4 +220,206 @@ public boolean foundData() {
+                (programName != null) ? programName : ""));
+
+        bbattributes.add(new BlackboardAttribute(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DOMAIN,
+                RecentActivityExtracterModuleFactory.getModuleName(),
+                (domain != null) ? domain : ""));
+
+        return bbattributes;
+    }
+
+    /**
+     * Creates a list of bookmark attributes from the passed in parameters.
+     *
+     * @param url Bookmark url
+     * @param title Title of the bookmarked page
+     * @param creationTime Date & time at which the bookmark was created
+     * @param programName Name of the module creating the attribute
+     * @param domain The domain of the bookmark's url
+     * @return A collection of bookmark attributes
+     */
+    protected Collection<BlackboardAttribute> createBookmarkAttributes(String url, String title, Long creationTime, String programName, String domain) {
+        Collection<BlackboardAttribute> bbattributes = new ArrayList<>();
+
+        bbattributes.add(new BlackboardAttribute(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_URL,
+                RecentActivityExtracterModuleFactory.getModuleName(),
+                (url != null) ? url : ""));
+
+        bbattributes.add(new BlackboardAttribute(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_TITLE,
+                RecentActivityExtracterModuleFactory.getModuleName(),
+                (title != null) ? title : ""));
+
+        if (creationTime != null) {
+            bbattributes.add(new BlackboardAttribute(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DATETIME_CREATED,
+                    RecentActivityExtracterModuleFactory.getModuleName(), creationTime));
+        }
+
+        bbattributes.add(new BlackboardAttribute(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PROG_NAME,
+                RecentActivityExtracterModuleFactory.getModuleName(),
+                (programName != null) ? programName : ""));
+
+        bbattributes.add(new BlackboardAttribute(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DOMAIN,
+                RecentActivityExtracterModuleFactory.getModuleName(),
+                (domain != null) ? domain : ""));
+
+        return bbattributes;
+    }
+
+     /**
+     * Creates a list of the attributes of a downloaded file
+     *
+     * @param path
+     * @param url URL of the downloaded file
+     * @param accessTime Time the download occurred
+     * @param domain Domain of the URL
+     * @param programName Name of the module creating the attribute
+     * @return A collection of attributed of a downloaded file
+     */
+    protected Collection<BlackboardAttribute> createDownloadAttributes(String path, Long pathID, String url, Long accessTime, String domain, String programName) {
+        Collection<BlackboardAttribute> bbattributes = new ArrayList<>();
+
+        bbattributes.add(new BlackboardAttribute(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PATH,
+                RecentActivityExtracterModuleFactory.getModuleName(),
+                (path != null) ? path : ""));
```

Comment:
// NON-NLS

Reply:
The //NON-NLS is required for the empty quotes?

Reply:
I am not sure, I imagine it depends on the localization candidate parser, but putting the tag comments on relieves us of having to know at the cost of some tedium :).